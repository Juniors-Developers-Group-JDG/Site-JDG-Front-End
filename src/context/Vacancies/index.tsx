'use client'

import { useFetch } from '@/hooks/useFetch'
import { Vacancy } from '@/types/Vacancy'
import { createContext, useEffect } from 'react'
import { toast } from 'react-toastify'

interface VacanciesContextData {
  vacancies: Vacancy[]
}

export const vacanciesContext = createContext<VacanciesContextData>(
  {} as VacanciesContextData,
)

interface VacanciesProviderProps {
  children: React.ReactNode
}

export function VacanciesProvider({ children }: VacanciesProviderProps) {
  const { data: vacancies = [], error } =
    useFetch<Vacancy[]>('/job-opportunities')

  useEffect(() => {
    if (error) {
      toast.error('Erro ao carregar as vagas!')
    }
  }, [error])

  return (
    <vacanciesContext.Provider value={{ vacancies }}>
      {children}
    </vacanciesContext.Provider>
  )
}
