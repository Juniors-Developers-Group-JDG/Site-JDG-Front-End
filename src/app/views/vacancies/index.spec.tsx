import Vacancies from './page'
import { render, screen } from '@testing-library/react'

describe('<Vacancies/>', () => {
  it('should render the VacanciesPage component', () => {
    render(<Vacancies />)

    const vacanciesPage = screen.getByTestId('vacancies-page')
    expect(vacanciesPage).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { asFragment } = render(<Vacancies />)

    expect(asFragment()).toMatchSnapshot()
  })
})
