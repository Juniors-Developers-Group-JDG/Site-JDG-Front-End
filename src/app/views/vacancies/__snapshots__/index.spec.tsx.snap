// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Vacancies/> should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="container flex flex-col items-center justify-center px-4 md:px-20"
    data-testid="vacancies-page"
  >
    <header
      class="mt-10 flex flex-col items-center justify-center"
      data-testid="vacancies-header-text"
    >
      <h1
        class="my-4 flex flex-col items-center justify-center text-xl font-bold text-primary-400 md:text-3xl"
      >
        Encontre sua vaga 
        <span>
          voluntária aqui!
        </span>
      </h1>
      <h2
        class="flex flex-col items-center justify-center text-xs text-white md:text-base"
      >
        Faça parte do Juniors Developers Group: 
        <span>
          onde o talento é voluntário e a inovação é infinita!
        </span>
      </h2>
    </header>
    <section
      class="mb-10 mt-20 flex w-full flex-col items-start justify-between gap-4 md:flex-row md:items-center"
    >
      <section
        class="relative"
      >
        <input
          class="w-full rounded-lg border border-primary-300 bg-primary-900 p-3 pl-12 text-start text-secondary outline-none transition-all ease-in placeholder:text-secondary focus:border-primary md:w-96"
          placeholder="Qual posição você está procurando?"
          type="text"
          value=""
        />
        <label
          class="absolute left-3 top-3"
        >
          <svg
            class="h-6 w-6 text-secondary"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 18a7.952 7.952 0 0 0 4.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0 0 18 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z"
            />
          </svg>
        </label>
      </section>
      <div
        class="relative inline-block text-left"
        data-headlessui-state=""
      >
        <section>
          <button
            aria-expanded="false"
            aria-haspopup="menu"
            class="inline-flex w-full items-center justify-center gap-x-1.5 rounded-md bg-primary-900 px-4 py-3 text-sm font-semibold  text-secondary shadow-sm hover:bg-zinc-900"
            data-headlessui-state=""
            id="headlessui-menu-button-:r1:"
            type="button"
          >
            <svg
              class="h-5 w-5"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7 11h10v2H7zM4 7h16v2H4zm6 8h4v2h-4z"
              />
            </svg>
            Filtrar por
            <svg
              aria-hidden="true"
              class="-mr-1 h-5 w-5 text-gray-400"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M5.23 7.21a.75.75 0 011.06.02L10 11.168l3.71-3.938a.75.75 0 111.08 1.04l-4.25 4.5a.75.75 0 01-1.08 0l-4.25-4.5a.75.75 0 01.02-1.06z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </section>
      </div>
    </section>
    <section
      class="flex w-full flex-col items-center justify-center md:flex-row md:flex-wrap md:justify-between"
    >
      <section
        class="min-w-96 my-2 h-full w-full rounded-lg border border-l-4 border-primary border-y-transparent border-r-transparent bg-primary-900 p-4 lg:w-[30rem]   "
      >
        <span
          class="rounded-lg bg-primary-800 p-2 text-xs text-secondary"
        >
          Estamos procurando um novo
        </span>
        <article
          class="mt-4"
        >
          <h1
            class="text-xl font-bold text-secondary"
          >
            Desenvolvedor Front-End
          </h1>
          <span
            class="relative p-2  text-xs text-primary-400"
            data-tech="HTML"
          >
            HTML
            <span
              class="absolute top-[5.5px] pl-[5.5px] text-primary-400"
            >
              .
            </span>
          </span>
          <span
            class="relative p-2  text-xs text-primary-400"
            data-tech="JavaScript"
          >
            JavaScript
            <span
              class="absolute top-[5.5px] pl-[5.5px] text-primary-400"
            >
              .
            </span>
          </span>
          <span
            class="relative p-2  text-xs text-primary-400"
            data-tech="React"
          >
            React
          </span>
          <p
            class="my-4 max-h-16 overflow-hidden text-xs text-secondary-500"
          >
            Lorem ipsum dolor sit, amet consectetur adipisicing elit. Minus minima ex sit fugiat. Sit aliquam minima ad similique quae obcaecati sunt, delectus eos excepturi mollitia ducimus ex at quam laudantium. Lorem ipsum dolor sit amet consectetur, adipisicing elit. Quis, ipsam molestiae natus quaerat aut totam voluptates, non explicabo rem fuga, provident nihil magni minima saepe optio animi ex quasi.  adipisicing elit. Quis, ipsam molestiae natus quaerat aut totam voluptates, non explicabo rem fuga, provident nihil magni minima saepe optio animi ex quasi. Error
          </p>
          <button
            class="bg-gradient-btn mt-4 h-12 w-full rounded-lg border border-primary text-base text-secondary outline-none transition-all ease-in-out hover:bg-primary-400 hover:font-semibold hover:text-secondary-800"
          >
            Voluntariar-se
          </button>
        </article>
      </section>
      <section
        class="min-w-96 my-2 h-full w-full rounded-lg border border-l-4 border-primary border-y-transparent border-r-transparent bg-primary-900 p-4 lg:w-[30rem]   "
      >
        <span
          class="rounded-lg bg-primary-800 p-2 text-xs text-secondary"
        >
          Estamos procurando um novo
        </span>
        <article
          class="mt-4"
        >
          <h1
            class="text-xl font-bold text-secondary"
          >
            Desenvolvedor Front-End
          </h1>
          <span
            class="relative p-2  text-xs text-primary-400"
            data-tech="HTML"
          >
            HTML
            <span
              class="absolute top-[5.5px] pl-[5.5px] text-primary-400"
            >
              .
            </span>
          </span>
          <span
            class="relative p-2  text-xs text-primary-400"
            data-tech="JavaScript"
          >
            JavaScript
            <span
              class="absolute top-[5.5px] pl-[5.5px] text-primary-400"
            >
              .
            </span>
          </span>
          <span
            class="relative p-2  text-xs text-primary-400"
            data-tech="React"
          >
            React
          </span>
          <p
            class="my-4 max-h-16 overflow-hidden text-xs text-secondary-500"
          >
            Lorem ipsum dolor sit, amet consectetur adipisicing elit. Minus minima ex sit fugiat. Sit aliquam minima ad similique quae obcaecati sunt, delectus eos excepturi mollitia ducimus ex at quam laudantium. Lorem ipsum dolor sit amet consectetur, adipisicing elit. Quis, ipsam molestiae natus quaerat aut totam voluptates, non explicabo rem fuga, provident nihil magni minima saepe optio animi ex quasi.  adipisicing elit. Quis, ipsam molestiae natus quaerat aut totam voluptates, non explicabo rem fuga, provident nihil magni minima saepe optio animi ex quasi. Error
          </p>
          <button
            class="bg-gradient-btn mt-4 h-12 w-full rounded-lg border border-primary text-base text-secondary outline-none transition-all ease-in-out hover:bg-primary-400 hover:font-semibold hover:text-secondary-800"
          >
            Voluntariar-se
          </button>
        </article>
      </section>
      <section
        class="min-w-96 my-2 h-full w-full rounded-lg border border-l-4 border-primary border-y-transparent border-r-transparent bg-primary-900 p-4 lg:w-[30rem]   "
      >
        <span
          class="rounded-lg bg-primary-800 p-2 text-xs text-secondary"
        >
          Estamos procurando um novo
        </span>
        <article
          class="mt-4"
        >
          <h1
            class="text-xl font-bold text-secondary"
          >
            Desenvolvedor Full-Stack
          </h1>
          <span
            class="relative p-2  text-xs text-primary-400"
            data-tech="HTML"
          >
            HTML
            <span
              class="absolute top-[5.5px] pl-[5.5px] text-primary-400"
            >
              .
            </span>
          </span>
          <span
            class="relative p-2  text-xs text-primary-400"
            data-tech="JavaScript"
          >
            JavaScript
            <span
              class="absolute top-[5.5px] pl-[5.5px] text-primary-400"
            >
              .
            </span>
          </span>
          <span
            class="relative p-2  text-xs text-primary-400"
            data-tech="React"
          >
            React
          </span>
          <p
            class="my-4 max-h-16 overflow-hidden text-xs text-secondary-500"
          >
            Lorem ipsum dolor sit, amet consectetur adipisicing elit. Minus minima ex sit fugiat. Sit aliquam minima ad similique quae obcaecati sunt, delectus eos excepturi mollitia ducimus ex at quam laudantium. Lorem ipsum dolor sit amet consectetur, adipisicing elit. Quis, ipsam molestiae natus quaerat aut totam voluptates, non explicabo rem fuga, provident nihil magni minima saepe optio animi ex quasi.  adipisicing elit. Quis, ipsam molestiae natus quaerat aut totam voluptates, non explicabo rem fuga, provident nihil magni minima saepe optio animi ex quasi. Error
          </p>
          <button
            class="bg-gradient-btn mt-4 h-12 w-full rounded-lg border border-primary text-base text-secondary outline-none transition-all ease-in-out hover:bg-primary-400 hover:font-semibold hover:text-secondary-800"
          >
            Voluntariar-se
          </button>
        </article>
      </section>
      <section
        class="min-w-96 my-2 h-full w-full rounded-lg border border-l-4 border-primary border-y-transparent border-r-transparent bg-primary-900 p-4 lg:w-[30rem]   "
      >
        <span
          class="rounded-lg bg-primary-800 p-2 text-xs text-secondary"
        >
          Estamos procurando um novo
        </span>
        <article
          class="mt-4"
        >
          <h1
            class="text-xl font-bold text-secondary"
          >
            Desenvolvedor Full-Stack
          </h1>
          <span
            class="relative p-2  text-xs text-primary-400"
            data-tech="HTML"
          >
            HTML
            <span
              class="absolute top-[5.5px] pl-[5.5px] text-primary-400"
            >
              .
            </span>
          </span>
          <span
            class="relative p-2  text-xs text-primary-400"
            data-tech="JavaScript"
          >
            JavaScript
            <span
              class="absolute top-[5.5px] pl-[5.5px] text-primary-400"
            >
              .
            </span>
          </span>
          <span
            class="relative p-2  text-xs text-primary-400"
            data-tech="React"
          >
            React
          </span>
          <p
            class="my-4 max-h-16 overflow-hidden text-xs text-secondary-500"
          >
            Lorem ipsum dolor sit, amet consectetur adipisicing elit. Minus minima ex sit fugiat. Sit aliquam minima ad similique quae obcaecati sunt, delectus eos excepturi mollitia ducimus ex at quam laudantium. Lorem ipsum dolor sit amet consectetur, adipisicing elit. Quis, ipsam molestiae natus quaerat aut totam voluptates, non explicabo rem fuga, provident nihil magni minima saepe optio animi ex quasi.  adipisicing elit. Quis, ipsam molestiae natus quaerat aut totam voluptates, non explicabo rem fuga, provident nihil magni minima saepe optio animi ex quasi. Error
          </p>
          <button
            class="bg-gradient-btn mt-4 h-12 w-full rounded-lg border border-primary text-base text-secondary outline-none transition-all ease-in-out hover:bg-primary-400 hover:font-semibold hover:text-secondary-800"
          >
            Voluntariar-se
          </button>
        </article>
      </section>
      <section
        class="min-w-96 my-2 h-full w-full rounded-lg border border-l-4 border-primary border-y-transparent border-r-transparent bg-primary-900 p-4 lg:w-[30rem]   "
      >
        <span
          class="rounded-lg bg-primary-800 p-2 text-xs text-secondary"
        >
          Estamos procurando um novo
        </span>
        <article
          class="mt-4"
        >
          <h1
            class="text-xl font-bold text-secondary"
          >
            Desenvolvedor Back-End
          </h1>
          <span
            class="relative p-2  text-xs text-primary-400"
            data-tech="HTML"
          >
            HTML
            <span
              class="absolute top-[5.5px] pl-[5.5px] text-primary-400"
            >
              .
            </span>
          </span>
          <span
            class="relative p-2  text-xs text-primary-400"
            data-tech="JavaScript"
          >
            JavaScript
            <span
              class="absolute top-[5.5px] pl-[5.5px] text-primary-400"
            >
              .
            </span>
          </span>
          <span
            class="relative p-2  text-xs text-primary-400"
            data-tech="React"
          >
            React
          </span>
          <p
            class="my-4 max-h-16 overflow-hidden text-xs text-secondary-500"
          >
            Lorem ipsum dolor sit, amet consectetur adipisicing elit. Minus minima ex sit fugiat. Sit aliquam minima ad similique quae obcaecati sunt, delectus eos excepturi mollitia ducimus ex at quam laudantium. Lorem ipsum dolor sit amet consectetur, adipisicing elit. Quis, ipsam molestiae natus quaerat aut totam voluptates, non explicabo rem fuga, provident nihil magni minima saepe optio animi ex quasi.  adipisicing elit. Quis, ipsam molestiae natus quaerat aut totam voluptates, non explicabo rem fuga, provident nihil magni minima saepe optio animi ex quasi. Error
          </p>
          <button
            class="bg-gradient-btn mt-4 h-12 w-full rounded-lg border border-primary text-base text-secondary outline-none transition-all ease-in-out hover:bg-primary-400 hover:font-semibold hover:text-secondary-800"
          >
            Voluntariar-se
          </button>
        </article>
      </section>
      <section
        class="min-w-96 my-2 h-full w-full rounded-lg border border-l-4 border-primary border-y-transparent border-r-transparent bg-primary-900 p-4 lg:w-[30rem]   "
      >
        <span
          class="rounded-lg bg-primary-800 p-2 text-xs text-secondary"
        >
          Estamos procurando um novo
        </span>
        <article
          class="mt-4"
        >
          <h1
            class="text-xl font-bold text-secondary"
          >
            Desenvolvedor Mobile
          </h1>
          <span
            class="relative p-2  text-xs text-primary-400"
            data-tech="HTML"
          >
            HTML
            <span
              class="absolute top-[5.5px] pl-[5.5px] text-primary-400"
            >
              .
            </span>
          </span>
          <span
            class="relative p-2  text-xs text-primary-400"
            data-tech="JavaScript"
          >
            JavaScript
            <span
              class="absolute top-[5.5px] pl-[5.5px] text-primary-400"
            >
              .
            </span>
          </span>
          <span
            class="relative p-2  text-xs text-primary-400"
            data-tech="React"
          >
            React
          </span>
          <p
            class="my-4 max-h-16 overflow-hidden text-xs text-secondary-500"
          >
            Lorem ipsum dolor sit, amet consectetur adipisicing elit. Minus minima ex sit fugiat. Sit aliquam minima ad similique quae obcaecati sunt, delectus eos excepturi mollitia ducimus ex at quam laudantium. Lorem ipsum dolor sit amet consectetur, adipisicing elit. Quis, ipsam molestiae natus quaerat aut totam voluptates, non explicabo rem fuga, provident nihil magni minima saepe optio animi ex quasi.  adipisicing elit. Quis, ipsam molestiae natus quaerat aut totam voluptates, non explicabo rem fuga, provident nihil magni minima saepe optio animi ex quasi. Error
          </p>
          <button
            class="bg-gradient-btn mt-4 h-12 w-full rounded-lg border border-primary text-base text-secondary outline-none transition-all ease-in-out hover:bg-primary-400 hover:font-semibold hover:text-secondary-800"
          >
            Voluntariar-se
          </button>
        </article>
      </section>
      <section
        class="min-w-96 my-2 h-full w-full rounded-lg border border-l-4 border-primary border-y-transparent border-r-transparent bg-primary-900 p-4 lg:w-[30rem]   "
      >
        <span
          class="rounded-lg bg-primary-800 p-2 text-xs text-secondary"
        >
          Estamos procurando um novo
        </span>
        <article
          class="mt-4"
        >
          <h1
            class="text-xl font-bold text-secondary"
          >
            Desenvolvedor Full-Stack - Banco de Talentos
          </h1>
          <span
            class="relative p-2  text-xs text-primary-400"
            data-tech="HTML"
          >
            HTML
            <span
              class="absolute top-[5.5px] pl-[5.5px] text-primary-400"
            >
              .
            </span>
          </span>
          <span
            class="relative p-2  text-xs text-primary-400"
            data-tech="JavaScript"
          >
            JavaScript
            <span
              class="absolute top-[5.5px] pl-[5.5px] text-primary-400"
            >
              .
            </span>
          </span>
          <span
            class="relative p-2  text-xs text-primary-400"
            data-tech="React"
          >
            React
          </span>
          <p
            class="my-4 max-h-16 overflow-hidden text-xs text-secondary-500"
          >
            Lorem ipsum dolor sit, amet consectetur adipisicing elit. Minus minima ex sit fugiat. Sit aliquam minima ad similique quae obcaecati sunt, delectus eos excepturi mollitia ducimus ex at quam laudantium. Lorem ipsum dolor sit amet consectetur, adipisicing elit. Quis, ipsam molestiae natus quaerat aut totam voluptates, non explicabo rem fuga, provident nihil magni minima saepe optio animi ex quasi.  adipisicing elit. Quis, ipsam molestiae natus quaerat aut totam voluptates, non explicabo rem fuga, provident nihil magni minima saepe optio animi ex quasi. Error
          </p>
          <button
            class="bg-gradient-btn mt-4 h-12 w-full rounded-lg border border-primary text-base text-secondary outline-none transition-all ease-in-out hover:bg-primary-400 hover:font-semibold hover:text-secondary-800"
          >
            Voluntariar-se
          </button>
        </article>
      </section>
      <section
        class="min-w-96 my-2 h-full w-full rounded-lg border border-l-4 border-primary border-y-transparent border-r-transparent bg-primary-900 p-4 lg:w-[30rem]   "
      >
        <span
          class="rounded-lg bg-primary-800 p-2 text-xs text-secondary"
        >
          Estamos procurando um novo
        </span>
        <article
          class="mt-4"
        >
          <h1
            class="text-xl font-bold text-secondary"
          >
            Desenvolvedor UX/UI
          </h1>
          <span
            class="relative p-2  text-xs text-primary-400"
            data-tech="HTML"
          >
            HTML
            <span
              class="absolute top-[5.5px] pl-[5.5px] text-primary-400"
            >
              .
            </span>
          </span>
          <span
            class="relative p-2  text-xs text-primary-400"
            data-tech="JavaScript"
          >
            JavaScript
            <span
              class="absolute top-[5.5px] pl-[5.5px] text-primary-400"
            >
              .
            </span>
          </span>
          <span
            class="relative p-2  text-xs text-primary-400"
            data-tech="React"
          >
            React
          </span>
          <p
            class="my-4 max-h-16 overflow-hidden text-xs text-secondary-500"
          >
            Lorem ipsum dolor sit, amet consectetur adipisicing elit. Minus minima ex sit fugiat. Sit aliquam minima ad similique quae obcaecati sunt, delectus eos excepturi mollitia ducimus ex at quam laudantium. Lorem ipsum dolor sit amet consectetur, adipisicing elit. Quis, ipsam molestiae natus quaerat aut totam voluptates, non explicabo rem fuga, provident nihil magni minima saepe optio animi ex quasi.  adipisicing elit. Quis, ipsam molestiae natus quaerat aut totam voluptates, non explicabo rem fuga, provident nihil magni minima saepe optio animi ex quasi. Error
          </p>
          <button
            class="bg-gradient-btn mt-4 h-12 w-full rounded-lg border border-primary text-base text-secondary outline-none transition-all ease-in-out hover:bg-primary-400 hover:font-semibold hover:text-secondary-800"
          >
            Voluntariar-se
          </button>
        </article>
      </section>
      <section
        class="min-w-96 my-2 h-full w-full rounded-lg border border-l-4 border-primary border-y-transparent border-r-transparent bg-primary-900 p-4 lg:w-[30rem]   "
      >
        <span
          class="rounded-lg bg-primary-800 p-2 text-xs text-secondary"
        >
          Estamos procurando um novo
        </span>
        <article
          class="mt-4"
        >
          <h1
            class="text-xl font-bold text-secondary"
          >
            Desenvolvedor Back-End
          </h1>
          <span
            class="relative p-2  text-xs text-primary-400"
            data-tech="HTML"
          >
            HTML
            <span
              class="absolute top-[5.5px] pl-[5.5px] text-primary-400"
            >
              .
            </span>
          </span>
          <span
            class="relative p-2  text-xs text-primary-400"
            data-tech="JavaScript"
          >
            JavaScript
            <span
              class="absolute top-[5.5px] pl-[5.5px] text-primary-400"
            >
              .
            </span>
          </span>
          <span
            class="relative p-2  text-xs text-primary-400"
            data-tech="React"
          >
            React
          </span>
          <p
            class="my-4 max-h-16 overflow-hidden text-xs text-secondary-500"
          >
            Lorem ipsum dolor sit, amet consectetur adipisicing elit. Minus minima ex sit fugiat. Sit aliquam minima ad similique quae obcaecati sunt, delectus eos excepturi mollitia ducimus ex at quam laudantium. Lorem ipsum dolor sit amet consectetur, adipisicing elit. Quis, ipsam molestiae natus quaerat aut totam voluptates, non explicabo rem fuga, provident nihil magni minima saepe optio animi ex quasi.  adipisicing elit. Quis, ipsam molestiae natus quaerat aut totam voluptates, non explicabo rem fuga, provident nihil magni minima saepe optio animi ex quasi. Error
          </p>
          <button
            class="bg-gradient-btn mt-4 h-12 w-full rounded-lg border border-primary text-base text-secondary outline-none transition-all ease-in-out hover:bg-primary-400 hover:font-semibold hover:text-secondary-800"
          >
            Voluntariar-se
          </button>
        </article>
      </section>
    </section>
  </section>
</DocumentFragment>
`;
